src/js/hooks/cycleCount/useAllProductsTab.jsx

```diff
@@ -87,6 +89,44 @@ const useAllProductsTab = ({ filterParams }) => {
     filterParams,
   });
 
+  const {
+    selectRow,
+    isChecked,
+    selectHeaderCheckbox,
+    selectedRowsAmount,
+    headerCheckboxProps,
+  } = useTableCheckboxes();
+
+  const getProductIds = () => tableData.data.map((row) => row.product.id);
```

Comment:
I guess this doesn't even have to be a method, but could be a const, like:
```js
const productIds = useMemo(() => tableData.data.map(...), [<any dependency indicating the tableData changed>])
```
I don't know though how the re-rendering of methods is executed for such cases, whether the mapping would be proceeded in the background while re-rendering or not.